tests/core/query/test_relational_query.py

```diff
@@ -3,7 +3,7 @@
 import pytest
 from anndata import AnnData
 from spatialdata import get_values, match_table_to_element
-from spatialdata._core.query.relational_query import _locate_value, _ValueOrigin
+from spatialdata._core.query.relational_query import _locate_value, _ValueOrigin, join_sdata_spatialelement_table
 from spatialdata.models.models import TableModel
 
 
```

Comment:
should we remove immediately or first deprecation warning?

Reply:
I'd remove the test immediately; I'd put a deprecation warning on `match_table_to_element()` (which will call `match_element_to_table()`, which in turn will use the new join APIs).

Reply:
I would also remove right away `_filter_table_by_elements()`. Please in doing it have a final double check that all that that function was doing is now supported by the new join APIs. If not, we should address that.

Reply:
This can't be removed straight away as it is used current status of queries like bounding box queries where we join before adding new elements to sdata.